DNN Platform/Library/Security/Permissions/PermissionController.cs

```diff
@@ -67,27 +68,37 @@ public static string BuildPermissions(IList permissions, string permissionKey)
             return permissionsString;
         }
 
+        /// <inheritdoc cref="IPermissionService.GetPermissionsByFolder" />
         public static ArrayList GetPermissionsByFolder()
         {
-            return new ArrayList(GetPermissions().Where(p => p.PermissionCode == "SYSTEM_FOLDER").ToArray());
+            return new ArrayList(GetPermissionsByFolderEnumerable().ToArray());
         }
 
+        /// <inheritdoc cref="IPermissionService.GetPermissionsByPortalDesktopModule" />
         public static ArrayList GetPermissionsByPortalDesktopModule()
         {
-            return new ArrayList(GetPermissions().Where(p => p.PermissionCode == "SYSTEM_DESKTOPMODULE").ToArray());
+            return new ArrayList(GetPermissionsByPortalDesktopModuleEnumerable().ToArray());
         }
 
+        /// <inheritdoc cref="IPermissionService.GetPermissionsByTab" />
         public static ArrayList GetPermissionsByTab()
         {
-            return new ArrayList(GetPermissions().Where(p => p.PermissionCode == "SYSTEM_TAB").ToArray());
+            return new ArrayList(GetPermissionsByTabEnumerable().ToArray());
         }

```

Comment:
Let's make sure and deprecate these methods, since they're especially inefficient now (with copying from one collection to another)

Reply:
I've added the deprecations with `[DnnDeprecated]`. Is the way I've added the deprecations OK, or should I change the message?

Example:
```cs
[DnnDeprecated(10, 0, 0, $"Use {nameof(IPermissionService)}.{nameof(IPermissionService.GetDefinitionsByFolder)} instead.")]
```

Reply:
Looks good to me. @dnnsoftware/approvers are we targeting this for 10, or is this okay for a 9.x release?

Reply:
As long as it is non-breaking (I did not see any breaking changes here). I would prefer we do this in v9 so we can remove the deprecated stuff in v11. It is fine to do in 10 too but then we need to get to 12 to do the actual removal.